src/cc/bcc_syms.cc

```diff
@@ -397,6 +397,38 @@ void bcc_symbol_free_demangle_name(struct bcc_symbol *sym) {
     free(const_cast<char*>(sym->demangle_name));
 }
 
+void bcc_symbol_adjust_entry_point(struct bcc_symbol *sym) {
+  if (!sym)
+    return;
+
+#if defined(__powerpc64__) && __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+  /*
+   * The PowerPC 64-bit ELF v2 ABI says that the 3 most significant bits
+   * in the st_other field of the symbol table specifies the number of
+   * instructions between a function's Global Entry Point (GEP) and Local
+   * Entry Point (LEP).
+   */
+  switch (sym->other >> 5) {
```

Comment:
If this is all what this `other` thing does (adjusting offset), maybe do it in the iteration function so that we could still keep the (potentially user-facing) callback unchanged...

Reply:
There is a slight problem I am seeing if I do it in the iteration function. Upon printing the user stacks, the symbol offsets are all being reduced by the factor deduced from `other`.

For example, when we should be seeing a stack trace like
```
__GI___inet_pton+0x8 [libc-2.27.so]
gaih_inet.constprop.7+0x1040 [libc-2.27.so]
getaddrinfo+0x164 [libc-2.27.so]
[unknown] [ping]
generic_start_main.isra.0+0x138 [libc-2.27.so]
__libc_start_main+0xc4 [libc-2.27.so]
```
we instead end up with
```
__GI___inet_pton+0x0 [libc-2.27.so]
gaih_inet.constprop.7+0x1038 [libc-2.27.so]
getaddrinfo+0x15c [libc-2.27.so]
[unknown] [ping]
generic_start_main.isra.0+0x130 [libc-2.27.so]
__libc_start_main+0xbc [libc-2.27.so]
```
So, the symbol offsets are all off by 8. One way to prevent this would be to do the offset adjustment in the user-facing API like it was done previously as that is the only place we know whether to use a symbol name or a direct binary offset. This way, we can keep the original symbol offsets (aka `st_value`) intact and not mess up the stack trace entries.

I am wondering if there is a simpler way to do this. Your thoughts?